File: app/src/main/java/soup/movie/di/ui/DetailUiModule.java
package soup.movie.di.ui;

import dagger.Module;
import dagger.Provides;
import soup.movie.data.MoobRepository;
import soup.movie.di.scope.ActivityScope;
import soup.movie.settings.TheaterSetting;
import soup.movie.ui.detail.DetailContract;
import soup.movie.ui.detail.DetailPresenter;

@Module
public class DetailUiModule {

    @ActivityScope
    @Provides
    DetailContract.Presenter presenter(
            MoobRepository moobRepository,
            TheaterSetting theaterSetting) {
        return new DetailPresenter(moobRepository, theaterSetting);
    }
}

File: app/src/main/java/soup/movie/di/ui/DetailUiModule.kt
package soup.movie.di.ui

import androidx.lifecycle.ViewModel
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import soup.movie.di.key.ViewModelKey
import soup.movie.ui.detail.DetailViewModel

@Module
interface DetailUiModule {

    @Binds
    @IntoMap
    @ViewModelKey(DetailViewModel::class)
    fun bindDetailViewModel(viewModel: DetailViewModel): ViewModel
}
